tests/func/experiments/test_experiments.py

```diff
@@ -225,18 +217,17 @@ def test_extend_branch(tmp_dir, scm, dvc):
     scm.add(["dvc.yaml", "dvc.lock", "copy.py", "params.yaml", "metrics.yaml"])
     scm.commit("init")
 
-    dvc.reproduce(stage.addressing, experiment=True, params=["foo=2"])
-    exp_a = dvc.experiments.scm.get_rev()
+    results = dvc.experiments.run(stage.addressing, params=["foo=2"])
+    exp_a = first(results)
```

Comment:
This may also resolve the flaky test issue noted in #4653.

I think it may have been caused by some git-python specific race condition when using `exp.scm.get_rev()`. We can avoid that call now that `experiments.run` returns the experiment hash directly.